routes/events.js

```diff
@@ -0,0 +1,159 @@
+        const statusMessage =
+            `All ${successCount} out of ${events.length} events were accepted.`;
+        req.logger.log('debug/events', statusMessage);
+
+        // Only set response if it hasn't yet finished,
+        // i.e. hasty response was not requested.
+        if (!res.finished) {
+            res.statusMessage = statusMessage;
+            res.status(204);
+            res.end();
+        }
+    } else if (invalidCount === events.length) {
+        // All events were invalid: 400
+        const statusMessage = `${invalidCount} out of ${events.length} ` +
+            `events were invalid and not accepted.`;
+        req.logger.log(
+            'warn/events',
+            { invalid: results.invalid, message: statusMessage }
+        );
+
+        if (!res.finished) {
+            res.statusMessage = statusMessage;
+            res.status(400);
+            res.json({ invalid: results.invalid });
+        }
+    } else if (failureCount !== events.length) {
+        // Some successes, but also some failures (invalid or errored): 207
+        const statusMessage = `${results.success.length} out of ${events.length} ` +
+            `events were accepted, but ${failureCount} failed (${invalidCount} ` +
+            `invalid and ${errorCount} errored).`;
+
+        req.logger.log(
+            'warn/events',
+            { error: results.error, invalid: results.invalid, message: statusMessage }
+        );
+
+        if (!res.finished) {
+            req.logger.log('info/events', 'FINISHING REQ');
+
+            res.statusMessage = statusMessage;
+            res.status(207);
+            res.json({ invalid: results.invalid, error: results.error });
+        }
+    } else {
+        // All events had some failure eith at least one error
+        // (some might have been invalid): 500
+        const statusMessage = `${failureCount} out of ${events.length} ` +
+            `events had failures and were not accepted. (${invalidCount} ` +
+            `invalid and ${errorCount} errored).`;
+        req.logger.log(
+            'error/events',
+            { errors: results.error, message: statusMessage }
+        );
+
+        if (!res.finished) {
+            res.statusMessage = statusMessage;
+            res.status(500);
+            res.json({ invalid: results.invalid, error: results.error });
+        }
+    }
+}
+
+
+module.exports = async(appObj) => {
+
+    app = appObj;
+
+    // Instantiate Eventbus from app.conf.  If eventbus_factory_module, require it
+    // to create a custom Eventbus instance.  Otherwise, use the default-eventbus factory.
+    const eventbusFactoryModule = _.get(
+        app.conf, 'eventbus_factory_module', '../lib/factories/default-eventbus'
+    );
+    app.logger.log(
+        'info/events',
+        `Instantiating Eventbus from ${eventbusFactoryModule}`
+    );
+
+    const eventbus = await require(eventbusFactoryModule).factory(app.conf, app.logger._logger);
```

Comment:
I would like to discuss more the usage of pure bunyan logger, but it's a bigger bikeshed